cupy/cuda/memory.pyx

```diff
@@ -486,29 +485,60 @@ cdef class SingleDeviceMemoryPool:
     cpdef Py_ssize_t _round_size(self, Py_ssize_t size):
         """Round up the memory size to fit memory alignment of cudaMalloc."""
         unit = self._allocation_unit_size
-        return (((size + unit - 1) // unit) * unit)
+        return ((size + unit - 1) // unit) * unit
 
     cpdef Py_ssize_t _bin_index_from_size(self, Py_ssize_t size):
         """Get appropriate bins (_free) index from the memory size"""
         unit = self._allocation_unit_size
         return (size - 1) // unit
 
-    cpdef void _grow_free_if_necessary(self, Py_ssize_t size):
-        """Extend bins (_free) size if necessary"""
-        current_size = len(self._free)
-        if current_size >= size:
-            return
-        growth_size = size - current_size
-        growth = [set() for i in range(growth_size)]
-        self._free.extend(growth)
+    cpdef _append_to_free_list(self, Py_ssize_t size, chunk):
+        cdef int index, bin_index
+        cdef set free_list
+        bin_index = self._bin_index_from_size(size)
```

Comment:
Can we remove calculation of `bin_index` now by any chances?
Because `size` is actually a `rounded_size`, we can put it into _index_vector 🤔

Reply:
Oh. But we have to use `long long`. if we avoid `_bin_index_from_size`.

Reply:
Yeah, we have both a merit and demerit. hmm